### PR TITLE
Fix a few OpenGL Pixel History Bugs

### PR DESCRIPTION
**Pull request by orsonbraines**
## Description
- Filter out events that affect a mip layer that is not selected
- Fix an off by one replay error causing weird behaviour
- Fix framebuffer blitting when attached textures change during frame

### CODE REVIEW

File: renderdoc/driver/gl/gl_pixelhistory.cpp [hunk 1]
```diff
@@ -1198,7 +1243,7 @@ void QueryPrimitiveIdPerFragment(WrappedOpenGL *driver, GLReplay *replay,
 
     // we expect this value to be overwritten by the primitive id shader. if not
     // it will cause an assertion that all color values are the same to fail
-    driver->glClearColor(0.84f, 0.17f, 0.2f, 0.49f);
+    driver->glClearColor(0.0f, 0.17f, 0.2f, 0.49f);
```
Comment (baldurk) [inline renderdoc/driver/gl/gl_pixelhistory.cpp]: What's the reason for this change? As in the comment this is expected to be overwritten so is there a need to clear to 0?

**Review by baldurk — CHANGES_REQUESTED** [review]
Only one change stands out as unexpected and needing perhaps some explanation, otherwise this looks good to me.

File: renderdoc/driver/gl/gl_pixelhistory.cpp (hunk 1 above)
Comment (orsonbraines) [reply renderdoc/driver/gl/gl_pixelhistory.cpp]: When fixing the off by one replay errors, we thought that if there were another bug that caused the primitive id not to be written, it would make more sense for the user to see a 0 as the primitive I'd rather than a large number from interpreting the bits of 0.84 as an int.

Comment (baldurk) [reply renderdoc/driver/gl/gl_pixelhistory.cpp]: If that's the case then it's better to make sure the primitive ID gets set to `~0U` in error cases so that it's displayed in the UI as an unknown primitive. Returning 0 is actually worse than returning an obviously large and incorrect number, since 0 is a real primitive ID that will be present in every draw which could mean the user might well go down a wrong path based on bad information.
